docs/File-Formats.md

```diff
@@ -1310,3 +1311,63 @@ The cells contain treatment response values that can be postive and negative rea
 <tr><td>AEW541</td><td>Larotrectinib</td><td>TrkA/B/C inhibitor</td><td>https://en.wikipedia.org/wiki/Larotrectinib</td><td>>8</td><td>2.33</td><td>2.68</td></tr>
 <tr><td>AZD0530</td><td>Saracatinib</td><td>Src/Bcr-Abl inhibitor</td><td>https://en.wikipedia.org/wiki/Saracatinib</td><td>NA</td><td>>8</td><td>4.60</td></tr>
 </table>
+
+## Generic Assay
+Generic Assay is a two dimensional matrix generalized to capture non-genetic measurements per sample. Instead of a gene per row and a sample per column, a Generic Assay file contains a generic entity per row and a sample per column. A generic entity is defined by the data curator and generally means something other than a gene. Some examples include, treatment response or mutational signatures. For each generic entity - sample pair, a real number represents a captured measurement.
+
+### Generic Assay meta file
+The generic assay metadata file should contain the following fields:
+```
+cancer_study_identifier: Same value as specified in meta file of the study
+genetic_alteration_type: GENERIC_ASSAY
+generic_assay_type: <GENERIC_ASSAY_TYPE>, e.g., "TREATMENT" or "MUTATIONAL_SIGNATURE"
+datatype: LIMIT-VALUE
+stable_id: Any unique identifier using a combination of alphanumeric characters, _ and -
+profile_name: A name describing the analysis.
+profile_description: A description of the data processing done.
+data_filename: <name of generic assay data file>
+show_profile_in_analysis_tab: true
+generic_entity_meta_properties: A comma separate list of generic entity properties, e.g., "name,description"
+```
+
+Example:
+```
+cancer_study_identifier: study_es_0
+genetic_alteration_type: GENERIC_ASSAY
+generic_assay_type: MUTATIONAL_SIGNATURE
+datatype: LIMIT-VALUE
+stable_id: mutational_signature
+profile_name: data of mutational signature
+profile_description: Description of data of mutational signature
+data_filename: data_mutational_signature.txt
+show_profile_in_analysis_tab: true
+generic_entity_meta_properties: name,description
+```
+
+#### Note on `stable_id`
+For generic assay, meta files values for `stable_id` are not pre-registered in the cBioPortal implementation, but can be freely chosen by the user. Requirement is that each generic assay meta file is assigned a unique `stable_id` by the user. This requirement is validated by the data validation scripts.
+
+#### Note on `generic_entity_meta_properties`
+All meta properties must be specified in `generic_entity_meta_properties`. Also, all meta properties specified in `generic_entity_meta_properties` are required to be shown in the data file as same name columns. These fields are free text (e.g., 'name of genetic entity').
```

Comment:
Some minor rewording:

"All meta properties must be specified in the `generic_entity_meta_properties` field.  Every meta property listed here must appear as a column header in the corresponding data file."

I'm not sure why you mean by the last part.  That is not a valid metadata property.  Unless I misunderstand something, lets drop it.

"These fields are free text (e.g., 'name of genetic entity')."